[2025-03-02T19:55:11.960+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-02T19:55:11.988+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: solar_energy_crops_pipeline.extract_data manual__2025-03-02T19:42:17.153002+00:00 [queued]>
[2025-03-02T19:55:12.000+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: solar_energy_crops_pipeline.extract_data manual__2025-03-02T19:42:17.153002+00:00 [queued]>
[2025-03-02T19:55:12.002+0000] {taskinstance.py:2866} INFO - Starting attempt 2 of 2
[2025-03-02T19:55:12.021+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): extract_data> on 2025-03-02 19:42:17.153002+00:00
[2025-03-02T19:55:12.034+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=537) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-03-02T19:55:12.036+0000] {standard_task_runner.py:72} INFO - Started process 539 to run task
[2025-03-02T19:55:12.037+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'solar_energy_crops_pipeline', 'extract_data', 'manual__2025-03-02T19:42:17.153002+00:00', '--job-id', '12', '--raw', '--subdir', 'DAGS_FOLDER/mainDag.py', '--cfg-path', '/tmp/tmpndfn60d4']
[2025-03-02T19:55:12.041+0000] {standard_task_runner.py:105} INFO - Job 12: Subtask extract_data
[2025-03-02T19:55:12.106+0000] {task_command.py:467} INFO - Running <TaskInstance: solar_energy_crops_pipeline.extract_data manual__2025-03-02T19:42:17.153002+00:00 [running]> on host 8e6d95be62f1
[2025-03-02T19:55:12.216+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='solar_energy_crops_pipeline' AIRFLOW_CTX_TASK_ID='extract_data' AIRFLOW_CTX_EXECUTION_DATE='2025-03-02T19:42:17.153002+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-02T19:42:17.153002+00:00'
[2025-03-02T19:55:12.218+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-03-02T19:55:12.248+0000] {logging_mixin.py:190} INFO - Extracting data from API...
[2025-03-02T19:55:12.251+0000] {logging_mixin.py:190} INFO - Fetching data for year 1967
[2025-03-02T19:55:12.995+0000] {logging_mixin.py:190} INFO - Failed to fetch data for 1967: 429
[2025-03-02T19:55:12.996+0000] {logging_mixin.py:190} INFO - Sleeping for a while to respect API rate limits...
[2025-03-02T19:55:21.005+0000] {logging_mixin.py:190} INFO - Fetching data for year 1968
[2025-03-02T19:55:21.173+0000] {logging_mixin.py:190} INFO - Failed to fetch data for 1968: 429
[2025-03-02T19:55:21.173+0000] {logging_mixin.py:190} INFO - Sleeping for a while to respect API rate limits...
[2025-03-02T19:55:29.177+0000] {logging_mixin.py:190} INFO - Fetching data for year 1969
[2025-03-02T19:55:29.342+0000] {logging_mixin.py:190} INFO - Failed to fetch data for 1969: 429
[2025-03-02T19:55:29.343+0000] {logging_mixin.py:190} INFO - Sleeping for a while to respect API rate limits...
[2025-03-02T19:55:37.349+0000] {logging_mixin.py:190} INFO - Fetching data for year 1970
[2025-03-02T19:55:37.512+0000] {logging_mixin.py:190} INFO - Failed to fetch data for 1970: 429
[2025-03-02T19:55:37.513+0000] {logging_mixin.py:190} INFO - Sleeping for a while to respect API rate limits...
[2025-03-02T19:55:45.512+0000] {logging_mixin.py:190} INFO - Fetching data for year 1971
[2025-03-02T19:55:45.677+0000] {logging_mixin.py:190} INFO - Failed to fetch data for 1971: 429
[2025-03-02T19:55:45.678+0000] {logging_mixin.py:190} INFO - Sleeping for a while to respect API rate limits...
[2025-03-02T19:55:53.688+0000] {logging_mixin.py:190} INFO - Fetching data for year 1972
[2025-03-02T19:55:53.854+0000] {logging_mixin.py:190} INFO - Failed to fetch data for 1972: 429
[2025-03-02T19:55:53.855+0000] {logging_mixin.py:190} INFO - Sleeping for a while to respect API rate limits...
[2025-03-02T19:56:01.859+0000] {logging_mixin.py:190} INFO - Fetching data for year 1973
[2025-03-02T19:56:02.024+0000] {logging_mixin.py:190} INFO - Failed to fetch data for 1973: 429
[2025-03-02T19:56:02.025+0000] {logging_mixin.py:190} INFO - Sleeping for a while to respect API rate limits...
[2025-03-02T19:56:10.027+0000] {logging_mixin.py:190} INFO - Fetching data for year 1974
[2025-03-02T19:56:10.194+0000] {logging_mixin.py:190} INFO - Failed to fetch data for 1974: 429
[2025-03-02T19:56:10.196+0000] {logging_mixin.py:190} INFO - Sleeping for a while to respect API rate limits...
[2025-03-02T19:56:17.799+0000] {local_task_job_runner.py:346} WARNING - State of this instance has been externally set to failed. Terminating instance.
[2025-03-02T19:56:17.821+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-03-02T19:56:17.828+0000] {process_utils.py:132} INFO - Sending 15 to group 539. PIDs of all processes in the group: [539]
[2025-03-02T19:56:17.829+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 539
[2025-03-02T19:56:17.831+0000] {taskinstance.py:3093} ERROR - Received SIGTERM. Terminating subprocesses.
[2025-03-02T19:56:17.919+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-03-02T19:56:17.964+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=539, status='terminated', exitcode=0, started='19:55:11') (539) terminated with exit code 0
